ad9959/src/lib.rs

```diff
@@ -468,16 +448,16 @@ impl<I: Interface> Ad9959<I> {
 
         // The function for channel frequency is `f_out = FTW * f_s / 2^32`, where FTW is the
         // frequency tuning word and f_s is the system clock rate.
-        let tuning_word: u32 = ((frequency / self.system_clock_frequency())
-            * 1u64.wrapping_shl(32) as f32)
-            as u32;
+        let tuning_word: u32 =
+            ((frequency as f32 / self.system_clock_frequency())
+                * 1u64.wrapping_shl(32) as f32) as u32;
 
         self.modify_channel(
             channel,
             Register::CFTW0,
             &tuning_word.to_be_bytes(),
         )?;
-        Ok((tuning_word as f32 / 1u64.wrapping_shl(32) as f32)
+        Ok((tuning_word as f32 / (1u64 << 32) as f32)
```

Comment:
I'd prefer to be explicit that we're doing a wrapping_shl() call. I've had shifts be implicitly modified due to compiler semantics before, and explicit is better than implicit.